src/QSymbolicsBase/rules.jl

```diff
@@ -38,9 +56,84 @@ function isnotflat_precheck(*)
 end
 
-FLATTEN_RULES = [
+##
+# Simplification rules
+## 
+
+# Flattening expressions
+RULES_FLATTEN = [
     @rule(~x::isnotflat_precheck(⊗) => flatten_term(⊗, ~x)),
     @rule ⊗(~~xs::hasscalings) => prefactorscalings_rule(xs)
 ]
 
-tensor_simplify = Fixpoint(Chain(FLATTEN_RULES))
+# Pauli identities
+RULES_PAULI = [
+    @rule(~o1::_isa(XGate)*~o2::_isa(XGate) => I),
+    @rule(~o1::_isa(YGate)*~o2::_isa(YGate) => I),
+    @rule(~o1::_isa(ZGate)*~o2::_isa(ZGate) => I),
+    @rule(~o1::_isa(XGate)*~o2::_isa(YGate) => im*Z),
+    @rule(~o1::_isa(YGate)*~o2::_isa(ZGate) => im*X),
+    @rule(~o1::_isa(ZGate)*~o2::_isa(XGate) => im*Y),
+    @rule(~o1::_isa(YGate)*~o2::_isa(XGate) => -im*Z),
+    @rule(~o1::_isa(ZGate)*~o2::_isa(YGate) => -im*X),
+    @rule(~o1::_isa(XGate)*~o2::_isa(ZGate) => -im*Y),
+    @rule(~o1::_isa(HGate)*~o2::_isa(XGate)*~o3::_isa(HGate) => Z),
+    @rule(~o1::_isa(HGate)*~o2::_isa(YGate)*~o3::_isa(HGate) => -Y),
+    @rule(~o1::_isa(HGate)*~o2::_isa(ZGate)*~o3::_isa(HGate) => X)
+]
+
+# Commutator identities
+RULES_COMMUTATOR = [
+    @rule(commutator(~o1::_isa(XGate), ~o2::_isa(YGate)) => 2*im*Z),
+    @rule(commutator(~o1::_isa(YGate), ~o2::_isa(ZGate)) => 2*im*X),
+    @rule(commutator(~o1::_isa(ZGate), ~o2::_isa(XGate)) => 2*im*Y),
+    @rule(commutator(~o1::_isa(YGate), ~o2::_isa(XGate)) => -2*im*Z),
+    @rule(commutator(~o1::_isa(ZGate), ~o2::_isa(YGate)) => -2*im*X),
+    @rule(commutator(~o1::_isa(XGate), ~o2::_isa(ZGate)) => -2*im*Y)
+]
+
+# Anticommutator identities
+RULES_ANTICOMMUTATOR = [
+    @rule(anticommutator(~o1::_isa(XGate), ~o2::_isa(XGate)) => 2*I),
+    @rule(anticommutator(~o1::_isa(YGate), ~o2::_isa(YGate)) => 2*I),
+    @rule(anticommutator(~o1::_isa(ZGate), ~o2::_isa(ZGate)) => 2*I),
+    @rule(anticommutator(~o1::_isa(XGate), ~o2::_isa(YGate))=> 0),
+    @rule(anticommutator(~o1::_isa(YGate), ~o2::_isa(ZGate)) => 0),
+    @rule(anticommutator(~o1::_isa(ZGate), ~o2::_isa(XGate)) => 0),
+    @rule(anticommutator(~o1::_isa(YGate), ~o2::_isa(XGate)) => 0),
+    @rule(anticommutator(~o1::_isa(ZGate), ~o2::_isa(YGate)) => 0),
+    @rule(anticommutator(~o1::_isa(XGate), ~o2::_isa(ZGate)) => 0)
+]
+
+RULES_ALL = [RULES_PAULI; RULES_COMMUTATOR; RULES_ANTICOMMUTATOR]
+
+##
+# Rewriters
+##
+qsimplify_flatten = Chain(RULES_FLATTEN)
+qsimplify_anticommutator = Chain(RULES_ANTICOMMUTATOR)
+qsimplify_pauli = Chain(RULES_PAULI)
+qsimplify_commutator = Chain(RULES_COMMUTATOR)
+
+"""Manually simplify a symbolic expression of quantum objects. 
+
+If the keyword `rewriter` is not specified, then `qsimplify` will apply every defined rule to the expression. 
+For performance or single-purpose motivations, the user has the option to define a specific rewriter for `qsimplify` to apply to the expression.
+
+```jldoctest
+julia> qsimplify(anticommutator(σˣ, σˣ), rewriter=qsimplify_anticommutator)
+2𝕀
+```
+"""
+function qsimplify(s; rewriter=nothing)
+    if QuantumSymbolics.isexpr(s)
+        if rewriter == nothing
```

Comment:
another stylistic note, it is more idiomatic to write `isnothing(rewriter)` or `rewriter === nothing`. Usually this does not matter because the compiler does "constant propagation" and basically ends up with this anyway, but it is a good habit to keep because `isnothing` and `===` guarantee the checks get compiled away when possible. The term to google about this is `type stability` and `specialization`